scalahost/nsc/src/main/scala/scala/meta/internal/databases/DenotationOps.scala

```diff
@@ -0,0 +1,101 @@
+      if (gsym.isConstructor && !gsym.isPrimaryConstructor) flags |= mf.SECONDARYCTOR
+      if (gsym.hasFlag(gf.MACRO)) flags |= mf.MACRO
+      if (gsym.isType && !gsym.isClass && !gsym.hasFlag(gf.PARAM)) flags |= mf.TYPE
+      if (gsym.isTerm && (gsym.hasFlag(gf.PARAM) || gsym.hasFlag(gf.PARAMACCESSOR)))
+        flags |= mf.PARAM
+      if (gsym.isType && gsym.hasFlag(gf.PARAM)) flags |= mf.TYPEPARAM
+      if (isObject) flags |= mf.OBJECT
+      if (gsym.hasFlag(gf.PACKAGE)) flags |= mf.PACKAGE
+      if (gsym.isModule && gsym.name == nme.PACKAGE) flags |= mf.PACKAGEOBJECT
+      if (gsym.isClass && !gsym.hasFlag(gf.TRAIT)) flags |= mf.CLASS
+      if (gsym.isClass && gsym.hasFlag(gf.TRAIT)) flags |= mf.TRAIT
+      if (maybeValOrVar && (gsym.hasFlag(gf.MUTABLE) || nme.isSetterName(gsym.name)))
+        flags |= mf.VAR
+      if (maybeValOrVar && !(gsym.hasFlag(gf.LOCAL) && gsym.hasFlag(gf.PARAMACCESSOR)))
+        flags |= mf.VAL
+      flags
+    }
+
+    private def accessQualifierFlags: Long = {
+      var flags = 0l
+      val gpriv = gsym.privateWithin.orElse(gsym.owner)
+      if (gsym.hasFlag(gf.SYNTHETIC) && gsym.hasFlag(gf.ARTIFACT)) {
+        // NOTE: some sick artifact vals produced by mkPatDef can be private to method (whatever that means)
+        // so here we just ignore them
+      } else {
+        if (gsym.hasFlag(gf.PROTECTED)) flags |= mf.PROTECTED
+        if (gsym.hasFlag(gf.PRIVATE) && !gsym.hasFlag(gf.PARAMACCESSOR)) flags |= mf.PRIVATE
+        // TODO: `private[pkg] class C` doesn't have PRIVATE in its flags
+        // so we need to account for that!
+        if (gsym.hasAccessBoundary && gpriv != g.NoSymbol && !gsym.hasFlag(gf.PROTECTED))
+          flags |= mf.PRIVATE
+      }
+      flags
+    }
+
+    private def otherFlags: Long = {
+      var flags = 0l
+      val isDeclaredDeferred = gsym.hasFlag(gf.DEFERRED) && !gsym.hasFlag(gf.PARAM)
+      val isDeclaredAbstract = (gsym.hasFlag(gf.ABSTRACT) && !gsym.hasFlag(gf.TRAIT)) || gsym
+          .hasFlag(gf.ABSOVERRIDE)
+      if (isDeclaredDeferred || isDeclaredAbstract) flags |= mf.ABSTRACT
+      if ((gsym.hasFlag(gf.FINAL) && !gsym.hasFlag(gf.PACKAGE)) || isObject) flags |= mf.FINAL
+      if (gsym.hasFlag(gf.SEALED)) flags |= mf.SEALED
+      if (gsym.hasFlag(gf.IMPLICIT)) flags |= mf.IMPLICIT
+      if (gsym.hasFlag(gf.LAZY)) flags |= mf.LAZY
+      if (gsym.hasFlag(gf.CASE)) flags |= mf.CASE
+      if (gsym.isType && gsym.hasFlag(gf.CONTRAVARIANT)) flags |= mf.CONTRAVARIANT
+      if (gsym.isType && gsym.hasFlag(gf.COVARIANT)) flags |= mf.COVARIANT
+      // TODO: mf.INLINE
+      flags
+    }
+
+    private def flags: Long = {
+      definitionFlags | accessQualifierFlags | otherFlags
+    }
+
+    private def name: String = {
+      gsym.decodedName.toString
+    }
+
+    private def info: String = {
+      if (gsym.isClass || gsym.isModule) ""
+      else {
+        val sig = gsym.info.toString
```

Comment:
Yes, it is definitely possible. But for that we, of course, can't rely on toString :)